File: CamelCase Pattern Matching[29-May-2023]/Solution.java
 ArrayList<String> CamelCase(int N,String[] Dictionary,String Pattern){
        
        Arrays.sort(Dictionary);
        
        ArrayList<String> ans = new ArrayList<>();
        
        for (String word : Dictionary) {
            String st = "";
            boolean flag = false;
            for (char c : word.toCharArray()) {
                if (Character.isUpperCase(c)) {
                    st += c;
                }
                
                if (st.equals(Pattern)) {
                    flag = true;
                    break;
                }
            }
            
            if (flag) {
                ans.add(word);
            }
        }
        
        if (ans.isEmpty()) {
            ans.add("-1");
        }
        
        return ans;
        
    }

File: CamelCase Pattern Matching[29-May-2023]/Solution.py
class Solution:
    def CamelCase(self,N,Dictionary,Pattern):
        ans=[]
        
        for word in Dictionary:
            st=""
            flag=False
            for char in word:
                if(char>="A" and char<="Z"):
                    st+=char
            
                if(st==Pattern):
                    flag=True
                    break
            
              
            if(flag==True):  
                print(ans)  
                ans.append(word)
        
        if(len(ans)==0):
            ans.append("-1")
        
        return ans        